examples/preprocessing/plot_target_encoder_cross_val.py

```diff
@@ -0,0 +1,159 @@
+"""
+==========================================
+Target Encoder's Internal Cross Validation
+==========================================
+
+.. currentmodule:: sklearn.preprocessing
+
+The :class:`TargetEnocoder` replaces each category of a categorical feature with
+the mean of the target variable for that category. This method is useful
+in cases where there is a strong relationship between the categorical feature
+and the target. To prevent overfitting, :meth:`TargetEncoder.fit_transform` uses
+interval cross validation to encode the training data to be used by a downstream
+model. In this example, we demonstrate the importance of the cross validation
+procedure to prevent overfitting.
+"""
+
+# %%
+# Create Synthetic Dataset
+# ========================
+# For this example, we build a dataset with three categorical features: an informative
+# and feature with medium cardinality, an uninformative feature with medium cardinality,
+# and an uninformative feature with high cardinality. First, we generate the informative
+# feature:
+from sklearn.preprocessing import KBinsDiscretizer
+import numpy as np
+
+n_samples = 50_000
+
+rng = np.random.RandomState(42)
+y = rng.randn(n_samples)
+noise = 0.5 * rng.randn(n_samples)
+n_categories = 100
+
+kbins = KBinsDiscretizer(
+    n_bins=n_categories, encode="ordinal", strategy="uniform", random_state=rng
+)
+X_informative = kbins.fit_transform((y + noise).reshape(-1, 1))
+
+# Permute the feature to remove the information from the ordering
+permuted_categories = rng.permutation(n_categories)
+X_informative = permuted_categories[X_informative.astype(np.int32)]
+
+# %%
+# The uninformative feature with medium cardinality is generated by permuting the
+# informative feature and removing the relationship with the target:
+X_shuffled = rng.permutation(X_informative)
+
+# %%
+# The uninformative feature with high cardinality is generated that is independent of
+# the target variable. We will show that target encoding without cross validation will
+# cause catastrophic overfitting for the downstream regressor. These high cardinality
+# features are typically unique identifies for samples which should be generally be
+# removed from machine learning dataset. In this example, We generate them to show how
+# :class:`TargetEncoder`'s default cross validation behavior mitigates the overfitting
+# issue automatically.
+X_near_unique_categories = rng.choice(
+    int(0.9 * n_samples), size=n_samples, replace=True
+).reshape(-1, 1)
+
+# %%
+# Finally, we assemble the dataset and perform a train test split:
+from sklearn.model_selection import train_test_split
+import pandas as pd
+
+X = pd.DataFrame(
+    np.concatenate(
+        [X_informative, X_shuffled, X_near_unique_categories],
+        axis=1,
+    ),
+    columns=["informative", "shuffled", "near_unique"],
+)
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+
+# %%
+# Training a Ridge Regressors
```

Comment:
Not sure which is the correct name of the model "Ridge Regressors" or "Ridge Regressor". I always thought the latter. 

```suggestion
# Training a Ridge Regressor
```

If that is right, then in the intro paragraph we should also change to "we train a ridge regressor".